File: src/Java/BehavioralPatterns/Mediator/CommunicationHub/src/Mediators/BroadcastMediator.java
package Mediators;

import Participants.Participant;

import java.util.ArrayList;
import java.util.List;

public class BroadcastMediator implements IMediator {
    private List<Participant> participants;

    public BroadcastMediator() {
        this.participants = new ArrayList<>();
    }

    public void setParticipants(List<Participant> participants) {
        this.participants = participants;
    }

    @Override
    public void notify(Participant sender, Object senderArgs) {
        for (Participant participant : participants) {
            participant.receive(sender, senderArgs);
        }
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CommunicationHub/mediators/broadcastMediator.js

const Mediator = require("./mediator");

class BroadcastMediator extends Mediator {
  constructor() {
    super();
    this.participants = [];
  }

  get participants() {
    return this._participants;
  }

  set participants(value) {
    this._participants = value;
  }

  notify(sender, senderArgs) {
    this.participants.forEach(p => p.receive(sender, senderArgs));
  }
}

module.exports = BroadcastMediator;